Handle 'out' keyword argument in constructors if indices are given

caspervdw opened the pull request:
Closes #345

brendan-ward commented on pygeos/_geometry.pyx:
```diff
@@ -54,10 +54,26 @@ cdef char _set_xyz(GEOSContextHandle_t geos_handle, GEOSCoordSequence *seq, unsi
+    if out.dtype != object:
+        raise TypeError("out array dtype must be object")
+    if out.ndim != 1:
+        raise TypeError("out is not a one-dimensional array.") 
```
```suggestion
        raise TypeError("out must be a one-dimensional array.") 
```

brendan-ward commented on pygeos/creation.py:
```diff
@@ -64,7 +64,9 @@ def points(coords, y=None, z=None, indices=None, **kwargs):
     else:
-        return simple_geometries_1d(coords, indices, GeometryType.POINT)
+        return simple_geometries_1d(
+            coords, indices, GeometryType.POINT, out=kwargs.get("out")
```
Is there are a reason for not making `out` a top-level parameter for `points` (or other top-level functions below)?

caspervdw replied:
I added it and fixed the docstrings.

caspervdw commented:
Thanks for the review! I addressed your points.

> It seems like the dtype of `out` is intercepted by numpy instead of the validation you added here
> 
> ```
> >>> pg.points([[0,0]], out=np.empty(2))
> Traceback (most recent call last):
>   ...
> TypeError: ufunc 'points' output (typecode 'O') could not be coerced to provided output parameter (typecode 'd') according to the casting rule ''same_kind''
> ```
> 
> Whereas your validation does run when `indices` is provided:
> 
> ```
> >>> pg.points([[0,0]], indices=[0], out=np.empty(2))
> Traceback (most recent call last):
>   ...
>   File "pygeos/_geometry.pyx", line 66, in pygeos._geometry._check_out_array
>     raise TypeError("out array dtype must be object")
> TypeError: out array dtype must be object
> ```

This is what I would expect indeed. To be clear: the validation I added is only ran when providing `indices`. Only then the Cython code is touched. If `indices` are not provided, the code flows straight into the numpy ufunc in which case numpy validates the kwargs (`out` etc.)
